### ISSUE TITLE
[BUG] bring own DEM and water mask error message "_uncropped"

### ISSUE THREAD

**Issue by dbekaert** [aria-tools/ARIA-tools#238]
**Describe the bug**
when bringing own DEM it looks for file*_uncropped.vrt; which does not exist when bringing own DEM.

**To Reproduce**
1. Download dem from dem.py .e.g ISCE
2. run ariaExtract.py for a geometry layer using the downloaded dem.

**Comment by EJFielding** [comment]
Not only does `ariaExtract` not find the DEM that is provided by the user, it erases all the DEM files when it fails.

**Comment by dbekaert** [comment]
Good point @EJFielding; this would not be desired. It is likely related to the "uncropped"; which was meant to be the untouched file. We will look into this.

**Comment by EJFielding** [comment]
I find a similar bug when I try to specify the mask file to avoid downloading it again.
```
ariaExtract.py -f S1-GUNW-D-R-131-tops-20201030_20201024-160721-38516N_36802N-RR-d813-v2_0_3.nc -l all -m mask/watermask.msk.vrt
***Extract Product Function:***
All (1) GUNW products meet spatial bbox criteria.
Group GUNW products into spatiotemporally continuous interferograms.
All (1) interferograms are spatially continuous.
All layers are to be extracted, pass all keys.
Thread count specified for gdal multiprocessing = 2
Traceback (most recent call last):
  File "/Users/fielding/anaconda3/envs/ARIA-tools/lib/python3.8/site-packages/ARIAtools/mask_util.py", line 111, in prep_mask
    gdal.Warp(maskfilename, os.path.join(workdir,os.path.basename(maskfilename).split('.')[0]+'_uncropped.msk.vrt'), options=gdal.WarpOptions(format=outputFormat, cutlineDSName=prods_TOTbbox, outputBounds=bounds, width=arrshape[1], height=arrshape[0], multithread=True, options=['NUM_THREADS=%s'%(num_threads)]))
  File "/Users/fielding/anaconda3/envs/ARIA-tools/lib/python3.8/site-packages/osgeo/gdal.py", line 613, in Warp
    srcDSTab = [Open(srcDSOrSrcDSTab)]
  File "/Users/fielding/anaconda3/envs/ARIA-tools/lib/python3.8/site-packages/osgeo/gdal.py", line 3308, in Open
    return _gdal.Open(*args)
RuntimeError: ./mask/watermask_uncropped.msk.vrt: No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/fielding/anaconda3/envs/ARIA-tools/bin/ariaExtract.py", line 16, in <module>
    main(inps)
  File "/Users/fielding/anaconda3/envs/ARIA-tools/lib/python3.8/site-packages/ARIAtools/extractProduct.py", line 897, in main
    inps.mask = prep_mask([[item for sublist in [list(set(d['amplitude'])) for d in standardproduct_info.products[1] if 'amplitude' in d] for item in sublist], [item for sublist in [list(set(d['pair_name'])) for d in standardproduct_info.products[1] if 'pair_name' in d] for item in sublist]], inps.mask, standardproduct_info.bbox_file, prods_TOTbbox, proj, amp_thresh=inps.amp_thresh, arrshape=arrshape, workdir=inps.workdir, outputFormat=inps.outputFormat, num_threads=inps.num_threads)
  File "/Users/fielding/anaconda3/envs/ARIA-tools/lib/python3.8/site-packages/ARIAtools/mask_util.py", line 129, in prep_mask
    raise Exception('Failed to open user mask')
Exception: Failed to open user mask
```
It similarly deleted the previously downloaded mask when opening the mask failed. It takes quite a while to download the watermask so it is quite inefficient to have to redo the download each time.

**Comment by dbekaert** [comment]
fixed per #239

**Comment by EJFielding** [comment]
Thanks for the help to fix this @bbuzz31 and @dbekaert .
